scripts/deploy.js

```diff
@@ -91,8 +91,21 @@ async function deployAll(
 
     const interchainTokenServiceAddress = await getCreate3Address(create3Deployer.address, wallet, deploymentKey);
     const tokenManagerDeployer = await deployContract(wallet, 'TokenManagerDeployer', []);
+
+    // Deploy both token implementations
     const interchainToken = await deployContract(wallet, 'InterchainToken', [interchainTokenServiceAddress]);
+    const hyperliquidInterchainToken = await deployContract(wallet, 'HyperliquidInterchainToken', [interchainTokenServiceAddress]);
```

Comment:
deploy the appropriate one based on whether chain is hyperliquid or not. deployment scripts will do the same